mockwebserver/src/main/java/okhttp3/mockwebserver/SocketPolicy.java

```diff
@@ -94,5 +94,16 @@
    * Fail HTTP/2 requests without processing them by sending an {@linkplain
    * MockResponse#getHttp2ErrorCode() HTTP/2 error code}.
    */
-  RESET_STREAM_AT_START
+  RESET_STREAM_AT_START,
+
+  /**
+   * For HTTP/2 connections, send a
+   * <a href="https://tools.ietf.org/html/rfc7540#section-6.8">GOAWAY frame</a> immediately after
+   * the response, then close the connection. The connection will be closed after delaying for
+   * {@link MockResponse#getShutdownDelay(java.util.concurrent.TimeUnit)}. Use this to simulate a
+   * graceful HTTP/2 connection shutdown.
+   *
+   * <p>See {@link SocketPolicy} for reasons why this can cause test flakiness and how to avoid it.
+   */
+  GRACEFUL_SHUTDOWN_AT_END
```

Comment:
Good thought. So a "graceful" shutdown would be a combination of `DISCONNECT_AT_END` + setting the shutdown delay? Then there would be no need for this value.